src/layouts/docs/utils/mdx-components.ts

```diff
@@ -0,0 +1,48 @@
+import Columns from 'components/author-primitives/vault/columns'
+import InlineTag from 'components/author-primitives/vault/inline-tag'
+import Placement from 'components/author-primitives/shared/placement-table'
+import NestedNode from 'components/author-primitives/waypoint/nested-node'
+import makeDocsAnchor from 'components/docs-anchor'
+import defaultMdxComponents from '@hashicorp/platform-docs-mdx'
+
+// Asana task to discuss MDX custom components in dev-portal:
+// https://app.asana.com/0/1100423001970639/1201648132084613/f
```

Comment:
I think the issue of how to handle custom MDX components is a pretty big one, and one we should try to tackle sooner rather than later. I have some suggestions in comments below on the possibility of removing the need to call `defaultMdxComponents` with different `product` options. I think this would be a great start, and would be happy to tackle it as part of this PR to try to mitigate this problem early.

Let me know what you think of the proposed comments below @ashleemboyer @BRKalow 🙇